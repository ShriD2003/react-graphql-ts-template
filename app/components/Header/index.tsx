/**
 *
 * Header
 *
 */
import React from 'react';
import { Layout } from 'antd';
import styled from 'styled-components';
import { injectIntl } from 'react-intl';
import { fonts, colors } from '@themes/index';
import T from '@components/T';
import logo from '@images/icon-512x512.png';

const StyledHeader = styled(Layout.Header)`
  && {
    &.ant-layout-header {
      padding: 0 1rem;
      height: 7rem;
      padding-bottom: 1rem;
      align-items: center;
    }
    display: flex;
    justify-content: center;
    background-color: ${colors.primary};
  }
`;
const Logo = styled.img`
  height: 5rem;
  width: auto;
  margin-top: 1rem;
`;
const Title = styled(T)`
  && {
    margin-bottom: 0;
    ${fonts.dynamicFontSize(fonts.size.xRegular, 1, 0.5)};
    display: flex;
    align-self: center;
    color: ${colors.secondaryText};
  }
`;

function Header() {
  return (
    <StyledHeader data-testid="header">
      <Logo alt="logo" src={logo} />
      <Title type="heading" id="wednesday_solutions" />
    </StyledHeader>
  );
}

export default injectIntl(Header);
