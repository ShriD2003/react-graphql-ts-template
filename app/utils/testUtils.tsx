import React from 'react';
import { I18nProvider } from '@lingui/react';
import { i18n } from '@lingui/core';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Route, Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import history from './history';

export const renderWithIntl = (children: React.ReactNode) => {
  i18n.load(DEFAULT_LOCALE, translationMessages[DEFAULT_LOCALE]);
  i18n.activate(DEFAULT_LOCALE);

  return render(<I18nProvider i18n={i18n}> {children} </I18nProvider>);
};

export const getComponentStyles = (Component: React.FC<any>, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props)!.type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  return window.getComputedStyle(componentRoots[0]);
};

export const renderProvider = (children: React.ReactNode, { path }: { path?: string } = {}, renderFn = render) => {
  const store = configureStore({}).store;
  return renderFn(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider
          theme={{
            main: 'violet'
          }}
        >
          <Router history={history}>{path ? <Route path={path}>{children}</Route> : children}</Router>
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = <Data,>(ok: boolean, data: Data, error?: object) => ({
  ok,
  data,
  error
});
