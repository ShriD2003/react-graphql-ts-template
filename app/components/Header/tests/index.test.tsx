/**
 *
 * Tests for Header
 *
 */

import React from 'react';
import { renderProvider } from '@utils/testUtils';
import Header from '../index';
import { fireEvent } from 'react-testing-library';

describe('<Header />', () => {
  let clickSpy: jest.Mock;

  beforeAll(() => {
    clickSpy = jest.fn();
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<Header mobile={false} toggleSidebar={clickSpy} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain logo', () => {
    const { getAllByAltText } = renderProvider(<Header mobile={true} toggleSidebar={clickSpy} />);
    expect(getAllByAltText('logo').length).toBe(1);
  });

  it('should trigger toggleSidebar when clicked on menu icon', () => {
    const { getByTestId } = renderProvider(<Header mobile={true} toggleSidebar={clickSpy} />);
    fireEvent.click(getByTestId('menu-icon'));
    expect(clickSpy).toBeCalled();
  });
});
