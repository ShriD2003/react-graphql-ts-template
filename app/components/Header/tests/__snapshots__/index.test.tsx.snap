// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
<body>
  <div>
    <header
      class="ant-layout-header Header__StyledHeader-a7tfu0-0 bfRGtz"
      data-testid="header"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          class="Header__Logo-a7tfu0-1 gRnKxJ"
          src="IMAGE_MOCK"
        />
      </a>
      <p
        class="T__StyledText-gcrjip-0 gjTOrr Header__Title-a7tfu0-2 fdMcFf"
        data-testid="t"
      >
        Wednesday Solutions
      </p>
    </header>
  </div>
</body>
`;
